Sources/Afluent/Workers/ShareFromCache.swift

```diff
@@ -17,22 +19,45 @@ extension AsynchronousUnitOfWork {
         hasher.combine(line)
         hasher.combine(column)
         let key = hasher.finalize()
+
+        let workToReturn: CachedWork
+
         switch strategy {
             case .cacheUntilCompletionOrCancellation:
-                return cache.retrieveOrCreate(
-                    unitOfWork: handleEvents(
-                        receiveOutput: { [weak cache] _ in
-                            cache?.clearAsynchronousUnitOfWork(withKey: key)
-                        },
-                        receiveError: { [weak cache] _ in
-                            cache?.clearAsynchronousUnitOfWork(withKey: key)
-                        },
-                        receiveCancel: { [weak cache] in
-                            cache?.clearAsynchronousUnitOfWork(withKey: key)
-                        }
-                    ).share(),
-                    keyedBy: key)
+                workToReturn =
+                    cache.retrieveOrCreate(
+                        unitOfWork: handleEvents(
+                            receiveOutput: { [weak cache] _ in
+                                cache?.clearAsynchronousUnitOfWork(withKey: key)
+                            },
+                            receiveError: { [weak cache] _ in
+                                cache?.clearAsynchronousUnitOfWork(withKey: key)
+                            },
+                            receiveCancel: { [weak cache] in
+                                cache?.clearAsynchronousUnitOfWork(withKey: key)
+                            }
+                        ).share(),
+                        keyedBy: key) as! CachedWork  // force unwrap since share() returns an opaque type
```

Comment:
Perhaps we just erase to `AnyAsynchronousUnitOfWork` and avoid the force unwrap?

Reply:
Sure, I could make that change.

Reply:
Ok done!